source/dem/distributions.cc

```diff
@@ -68,3 +68,59 @@ UniformDistribution::find_max_diameter()
 {
   return this->diameter_values;
 }
+
+HistogramDistribution::HistogramDistribution(
+  const std::vector<double> &d_list,
+  const std::vector<double> &d_probabilities)
+{
+  diameter_hist_values = d_list;
+  std::vector<double> cumulative_probability_vector;
+  cumulative_probability_vector.reserve(d_probabilities.size());
```

Comment:
Is there a reason for the use of reserve?

```suggestion
  const unsigned int n_diameters = d_probabilities.size();
  std::vector<double> cumulative_probability_vector(n_diameters);
```

Reply:
Allocating memory once and not every time I push_back the vector.